c/stcbackground.c

```diff
@@ -0,0 +1,73 @@
+#include "logging.h"
+#include "stcbackground.h"
+
+static int processStcBackgroundHandler(STCBase *base, STCModule *module, int selectStatus) {
+  BackgroundModuleData* moduleData = (BackgroundModuleData*)(module->data);
+  BackgroundTask* taskList = moduleData->taskList;
+
+  for (int i = 1; i < N_TASK_TABLE_ENTRIES; i++) {
+    BackgroundTask* t = &taskList[i];
+    if (t->id != 0)  /* initialise */ {
+      t->countInterval+=STC_BACKGROUND_INTERVAL;
```

Comment:
You're assuming that the interval is always equal to `STC_BACKGROUND_INTERVAL`, which is not true.

Reply:
its part of logic when to process registered handler, when this counter goes to zero, we execute its associated background handler, and reset it to desired multiple again.
In short, this is what we needed.

Reply:
this is fine